tests/locallib_test.php

```diff
@@ -29,15 +29,92 @@
 require_once($CFG->dirroot . '/mod/accredible/locallib.php');
 
 class mod_accredible_locallib_testcase extends advanced_testcase {
-    /**
-     * Sample test
-     * later: remove after adding real tests
-     */
-    public function test_sample() {
+    function setUp(): void {
         $this->resetAfterTest();
-        $this->setAdminUser();
-    
-        // Sample test.
-        $this->assertEmpty("");
+        $this->user = $this->getDataGenerator()->create_user();
+        $this->course = $this->getDataGenerator()->create_course();
+    }
+
+    public function test_accredible_get_transcript() {
+        global $DB;
+
+        /**
+        * When no quiz available for user.
+        */
+        $result = accredible_get_transcript($this->course->id, $this->user->id, 5);
+
+        /**
+        * it responds with false
+        */
+        $this->assertEmpty($DB->get_records('quiz'));
+        $this->assertEquals($result, false);
+
+        /**
+         * When user has taken multiple quizzes.
+         */
+        $quiz1 = $this->createQuizModule($this->course->id);
+        $quiz2 = $this->createQuizModule($this->course->id);
+        $quiz3 = $this->createQuizModule($this->course->id);
+
+        /**
+        * When user has completed multiple quizzes and has a passing grade.
+        */
+        $this->createQuizGrades($quiz1->id, $this->user->id, 10);
+        $this->createQuizGrades($quiz2->id, $this->user->id, 5);
+        $this->createQuizGrades($quiz3->id, $this->user->id, 5);
+
+        $result = accredible_get_transcript($this->course->id, $this->user->id, 0);
```

Comment:
It would be better to also include the scenario in which the admin enables "Auto-issue criteria: by final quiz grade" in the Accredible activity. In this case, the third argument `$final_quiz_id` will be one of the quiz IDs in the course and the result will exclude the grade of the final quiz.